### PR TITLE
Notifications: Global preferences

### PR DESCRIPTION
**Pull request by rperez89**
Needs previous pr's because of the asset.

Screen Shot 2020-05-13 at 7 09 21 PM

Screen Shot 2020-05-13 at 7 09 34 PM

### CODE REVIEW

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                wide
+                mode="strong"
+                onClick={onLockSettings}
+              >
+                Lock settings
+              </Button>
+              <Button
+                css={`
+                  margin-top: ${2 * GU}px;
+                  justify-content: left;
```
Comment (fabriziovigevani) [inline src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:66]: Shouldn't this be centered? Same for `Delete your email` Button

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                `}
+                wide
+                onClick={onUpdateEmail}
+              >
+                <IconMail />
```
Comment (fabriziovigevani) [inline src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:71]: Should we use the `icon` prop for the `Button`?

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+import {
+  Box,
+  Button,
+  GU,
+  IconMail,
+  IconTrash,
+  Info,
+  Link,
+  Switch,
+  useTheme,
```
Comment (fabriziovigevani) [inline src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:11]: Could we order this one?

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+          `}
+        >
+          <Switch
+            checked={!notificationsDisabled}
+            onChange={handleSwitchNotificationsStatus}
+          />
+          <span
+            css={`
+              ${textStyle('body1')};
+              margin-left: ${GU}px;
```
Comment (fabriziovigevani) [inline src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:147]:
```suggestion
              margin-left: ${1 * GU}px;
```

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js (hunk 1 above)
Comment (rperez89) [reply src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:66]: no, if you center that  they don't look like on the designs

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js (hunk 4 above)
Comment (rperez89) [reply src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:147]: haha never understand why we are adding a 1, seen this comment several times haha. Why is that?

Comment (fabriziovigevani) [reply src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:147]: Haha, i think it's better to make explicit the number of grid units we are using even tough it's the same result.

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js (hunk 1 above)
Comment (fabriziovigevani) [reply src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:66]: Aren't they centered ? https://www.figma.com/file/IdVA6iMn0cWLV5imylwEUP/%E2%9A%96%EF%B8%8F-Court-Dashboard-Master?node-id=4980%3A1815

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+                `}
+                wide
+                onClick={onDeleteEmail}
+              >
+                <IconTrash
+                  css={`
+                    color: ${theme.negative};
+                  `}
+                />
+                Delete your email
```
Comment (fabriziovigevani) [inline src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:87]: Looking at the Figmas again i think this should be just `Delete email` ?

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js (hunk 4 above)
Comment (sohkai) [reply src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:147]: If/When we introduce [the babel preprocessor to transform the gu](https://github.com/aragon/aragon-ui/issues/333), we will have to use the `1 * GU` format, as with css, `1px` is not the same as just `px`.

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+  notificationsDisabled,
+  onSwitchNotificationsStatus,
+  onLockSettings,
+  onDeleteEmail,
+  onUpdateEmail,
+}) {
+  const theme = useTheme()
+
+  return (
+    <React.Fragment>
```
Comment (sohkai) [inline src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:30]: Nitpick: should we consolidate usage of `<React.Fragment>` and `<>`? I think there are a few other PRs just using `<>`.

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+          <React.Fragment>
+            <Box heading="Email account">
+              <Tag
+                uppercase={false}
+                background={theme.infoSurface}
+                size="normal"
+                color={theme.content}
+                css={`
+                  ${textStyle('body2')};
+                  border-radius: 3px;
```
Comment (sohkai) [inline src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:48]: I think this is meant to look like a Badge, so we should be `{RADIUS}px`.

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+                Delete your email
+              </Button>
+              <Info
+                css={`
+                  margin-top: ${2 * GU}px;
+                `}
+              >
+                Learn more about our{' '}
+                <Link href="https://aragon.one/email-collection.md">
+                  Email collection policy
```
Comment (sohkai) [inline src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:96]:
```suggestion
                  email collection policy
```

I think elsewhere we don't capitalize "email".

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+function EmailNotificationBox({
+  notificationsDisabled,
+  onSwitchNotificationsStatus,
+}) {
+  const handleSwitchNotificationsStatus = useCallback(() => {
+    onSwitchNotificationsStatus(!notificationsDisabled)
+  }, [notificationsDisabled, onSwitchNotificationsStatus])
+
+  return (
+    <Box heading="Signed In With Email">
```
Comment (sohkai) [inline src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:117]:
```suggestion
    <Box heading="Email notifications">
```

Not sure, but [the designs](https://www.figma.com/file/IdVA6iMn0cWLV5imylwEUP/%E2%9A%96%EF%B8%8F-Court-Dashboard-Master?node-id=4980%3A25193) appear to have this copy instead.

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+          `}
+          mode={notificationsDisabled ? 'warning' : 'info'}
+        >
+          {notificationsDisabled
+            ? `We strongly advise you to turn on these notifications. 
+            By turning off these email alerts, we won’t notify you whenever you are drafted to arbitrate a dispute or 
+            if you have any upcoming tasks. Failure to complete your jury duties will result in monetary penalties. `
+            : `We will timely notify you whenever you are drafted to arbitrate a
+          dispute, if you have any upcoming tasks and when you can start
+          claiming your rewards.`}
```
Comment (sohkai) [inline src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:165]:
```suggestion
            ? `We strongly advise you to enable these notifications. 
              By disabling these email alerts, you will not be notified when you are drafted to arbitrate a dispute or 
              when your upcoming tasks are due. Failure to complete your duties as a juror will result in monetary penalties.`
            : `We will notify you for any relevant Aragon Court events, such as when you are drafted to arbitrate a
              dispute, when your upcoming tasks are due, and when you have rewards to claim.`}
```

Updated the indenting and suggested an edit for the copy that reads better for me cc @dizzypaty.

File: src/components/GlobalPreferences/Notifications/NotificationsPreferences.js [hunk 11]
```diff
@@ -58,23 +60,38 @@ export default function NotificationsPreferences({
                 mode="strong"
                 onClick={onLockSettings}
               >
-                Lock settings
+                <IconLock />
+                <span
+                  css={`
+                    width: ${12 * GU}px;
```
Comment (sohkai) [inline src/components/GlobalPreferences/Notifications/NotificationsPreferences.js:66]: Not a huge deal, but setting widths on `spans` is not very common. We most likely want to set a width on the `Button` instead?

**Pull request merged**